### ISSUE TITLE
Branding Auth Pages

### ISSUE THREAD

**Issue by ruiengana** [JanssenProject/jans-auth-server#20]
When applying branding there are internal servlets to calculate location uri of custom logo and favicon. Are these servlets applicable for Jans Auth Server? How can we configure custom logos in Jans Auth Server?

**Comment by nynymike** [comment]
What exactly are you branding? The login pages?

**Comment by ruiengana** [comment]
Yes, the login and consent pages.

**Comment by ruiengana** [comment]
> What exactly are you branding? The login pages?
Yes, the login and consent pages.

**Comment by nynymike** [comment]
Sorry, not all the docs are moved over from Gluu yet. Did you review: https://gluu.org/docs/gluu-server/4.2/operation/custom-design/ ?

**Comment by ruiengana** [comment]
> Sorry, not all the docs are moved over from Gluu yet. Did you review: https://gluu.org/docs/gluu-server/4.2/operation/custom-design/ ?

Yes, and from Gluu current docs we should put all custom pages in custom which is fine, but when it comes to logos / favicons there is some internal servlet that fetch the config and depending if you have uploaded custom logos it returns custom URIs or default ones.

Since custom logo configuration feature is is mainly specifically to oxTrust, can we use internal servlets? How can we reference our logos now?

**Comment by HemantKMehta** [comment]
> > Sorry, not all the docs are moved over from Gluu yet. Did you review: https://gluu.org/docs/gluu-server/4.2/operation/custom-design/ ?
> 
> Yes, and from Gluu current docs we should put all custom pages in custom which is fine, but when it comes to logos / favicons there is some internal servlet that fetch the config and depending if you have uploaded custom logos it returns custom URIs or default ones.
> 
> Since custom logo configuration feature is is mainly specifically to oxTrust, can we use internal servlets? How can we reference our logos now?

Please also check this part of the above link 

All images should be placed under:
/opt/gluu/jetty/oxauth/custom/static/img
Note
You can change the logo on every public-facing page here. Place your image in /static/img and name it logo.png.

**Comment by ruiengana** [comment]
OK, and lines of code like '''<link rel="icon" href="#{request.contextPath}/servlet/favicon" type='image/x-icon' />''' will be left as-is or customised to '''<link rel="icon" href="#/img/favicon" type='image/x-icon' />''' ?

**Comment by ruiengana** [comment]
Finally managed to solve this by copying logo.png and favicon.ico into custom/static/img and then applying the following replaces in .xhtml files:
- '#{request.contextPath}/servlet/logo' for 'ext/resources/img/logo.png'
- '#{request.contextPath}/servlet/favicon' for 'ext/resources/img/favicon.ico'

NOTE: I still think references to #{request.contextPath}/servlet/logo and #{request.contextPath}/servlet/favicon in default pages should be removed as they are related to oxTrust. Not sure if this should be a fix or enhancement.

**Comment by HemantKMehta** [comment]
Sharing just for your information:  I see that these servlets are available in Janssen : 
/opt/jetty-9.4/temp/jetty-localhost-8081-jans-auth_war-_jans-auth-any-18272919089717614003.dir/webapp/WEB-INF/classes/io/jans/as/server/servlet$ ls
FaviconServlet.class            
LogoServlet.class    
They are referring to 
/opt/gluu/jetty/jans-auth/custom/static/favicon/ and 
/opt/gluu/jetty/jans-auth/custom/static/logo/ 
folders.

**Comment by ruiengana** [comment]
> this is just to confirm you that These servlets are available in Janssen :
> /opt/jetty-9.4/temp/jetty-localhost-8081-jans-auth_war-_jans-auth-any-18272919089717614003.dir/webapp/WEB-INF/classes/io/jans/as/server/servlet$ ls
> FaviconServlet.class
> LogoServlet.class
> Just like the Gluu Server here also they are referring to following folders
> /opt/gluu/jetty/jans-auth/custom/static/favicon/
> /opt/gluu/jetty/jans-auth/custom/static/logo/

So, recommendation is to keep using the servlets but put logo.png in custom/static/logo/ and favicon.ico in custom/static/favicon/ ?

**Comment by HemantKMehta** [comment]
Yes, that should work.

**Comment by ruiengana** [comment]
> Yes, that should work.

Hi @HemantKMehta , just tried and this doesn't seem to work. I still get default logos.

So, I am not customising `custom/pages` just simply adding my custom images into `custom/static/logo/logo.png` and `custom/static/favicon/favicon.ico`

Any ideas?

**Comment by HemantKMehta** [comment]
Yes adding the custom images to custom/static/logo/logo.png and custom/static/favicon/favicon.ico should work without changing the pages.

**Comment by ruiengana** [comment]
That was my test, it doesn't work :(

**Comment by HemantKMehta** [comment]
Ok thanks for the confirmation. I will try to do it on my setup and then update you

**Comment by nynymike** [comment]
This is a very important topic. Customization of logos and look and feel should be well documented. If the code design is wrong, now is a good time to fix it. Also, the final resolution should be documented on the wiki on https://jans.io

**Comment by yuriyz** [comment]
@HemantKMehta We must not hard code paths, please add configuration properties for it so path is configurable (with default values which refers to jans resources).
